src/main/java/Duke.java

```diff
@@ -1,10 +1,191 @@
+                break;
+            default:
+                promptUserInputInvalid();
+                break;
+            }
+        } while (isLoop);
+    }
+
+    private static void promptUserInputInvalid() {
+        System.out.println("Invalid command! Please try again!");
+    }
+
+    private static boolean executeCommandBye(String rawInput, String[] arguments) {
+        if (arguments.length != 0) {
+            System.out.println("Command \"Bye\" requires no argument. Please try again!");
+            return true;
+        }
+
+        quitProgram();
+        return false;
+    }
+
+    private static void executeCommandDone(String[] arguments) {
+        int targetRecordIndex = -1;
+        boolean isArgumentInteger = true;
+        try {
+            targetRecordIndex = Integer.parseInt(arguments[0]) - 1;
+        } catch (NumberFormatException e) {
+            isArgumentInteger = false;
+        }
+
+        if (arguments.length == 1 && isArgumentInteger) {
+            markAsDone(targetRecordIndex);
+        } else {
+            System.out.println("Command \"Done\" only requires an integer argument. Please try again!");
+        }
+    }
+
+    private static String getUserInput() {
+        Scanner scan = new Scanner(System.in);
+        String userInput = scan.nextLine();
+        System.out.println("Command entered: " + userInput);
+        return userInput;
+    }
+
+    private static void printWelcomeMsg() {
+        System.out.println("Hello! I am Kaman :)");
+        System.out.println("What can I do for you?");
+    }
+
+    private static void markAsDone(int index) {
+        if (index < 0 || index >= records.size()) {
+            System.out.println("Invalid input! (Index cannot be out of bounds)");
+            return;
+        }
+        records.get(index).setAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("\t" + records.get(index));
+    }
+
+    private static void addRecord(String[] detailFragments, String taskType) {
+        String[] details = null;
+        String taskName;
+        String date;
+        switch (taskType) {
+        case Todo.TASK_TYPE:
+            taskName = String.join(" ", detailFragments);
+            records.add(new Todo(taskName));
+            break;
+        case Deadline.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Deadline(taskName, date));
+            }
+            break;
+        case Event.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Event(taskName, date));
+            }
+            break;
+        }
```

Comment:
I believe a "default" case should be added here, so as to adhere to the coding standard on using switch statement.